rskj-core/src/main/java/co/rsk/rpc/netty/JsonRpcWeb3ServerHandler.java

```diff
@@ -51,29 +52,41 @@ public class JsonRpcWeb3ServerHandler extends SimpleChannelInboundHandler<ByteBu
     private final ObjectMapper mapper = new ObjectMapper();
     private final JsonNodeFactory jsonNodeFactory = JsonNodeFactory.instance;
     private final JsonRpcBasicServer jsonRpcServer;
+    private final int defaultTimeout;
 
     @VisibleForTesting
-    JsonRpcWeb3ServerHandler(JsonRpcBasicServer jsonRpcServer) {
+    JsonRpcWeb3ServerHandler(JsonRpcBasicServer jsonRpcServer, RskSystemProperties rskSystemProperties) {
         this.jsonRpcServer = jsonRpcServer;
+
+        this.defaultTimeout = rskSystemProperties.getRpcTimeout();
     }
 
-    public JsonRpcWeb3ServerHandler(Web3 service, List<ModuleDescription> filteredModules, int maxBatchRequestsSize) {
+    public JsonRpcWeb3ServerHandler(Web3 service, List<ModuleDescription> filteredModules, int maxBatchRequestsSize, RskSystemProperties rskSystemProperties) {
         this.jsonRpcServer = new JsonRpcBasicServer(service, service.getClass());
         List<JsonRpcInterceptor> interceptors = new ArrayList<>();
         interceptors.add(new JsonRpcRequestValidatorInterceptor(maxBatchRequestsSize));
         jsonRpcServer.setInterceptorList(interceptors);
         jsonRpcServer.setRequestInterceptor(new JsonRpcMethodFilter(filteredModules));
         jsonRpcServer.setErrorResolver(new MultipleErrorResolver(new RskErrorResolver(), AnnotationsErrorResolver.INSTANCE, DefaultErrorResolver.INSTANCE));
+
+        this.defaultTimeout = rskSystemProperties.getRpcTimeout();
     }
 
     @Override
     protected void channelRead0(ChannelHandlerContext ctx, ByteBufHolder request) throws Exception {
         ByteBuf responseContent = Unpooled.buffer();
         int responseCode;
+
         try (ByteBufOutputStream os = new ByteBufOutputStream(responseContent);
-             ByteBufInputStream is = new ByteBufInputStream(request.content().retain())) {
+             ByteBufInputStream is = new ByteBufInputStream(request.content().retain());
+             ExecTimeoutContext ignored = ExecTimeoutContext.create(defaultTimeout)) {
 
-            responseCode = jsonRpcServer.handleRequest(is, os);
+            if (defaultTimeout <= 0) {
+                responseCode = jsonRpcServer.handleRequest(is, os);
+            } else {
+                responseCode = jsonRpcServer.handleRequest(is, os);
+                ExecTimeoutContext.checkIfExpired();
+            }
         } catch (JsonRpcRequestPayloadException e) {
```

Comment:
don't we need one more catch block for `TimeoutException` here?

Reply:
Done